agent/connect/ca/provider_vault_test.go

```diff
@@ -16,11 +17,16 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+var vaultLock sync.Mutex
+
 func testVaultCluster(t *testing.T) (*VaultProvider, *vault.Core, net.Listener) {
 	return testVaultClusterWithConfig(t, true, nil)
 }
 
 func testVaultClusterWithConfig(t *testing.T, isRoot bool, rawConf map[string]interface{}) (*VaultProvider, *vault.Core, net.Listener) {
+	vaultLock.Lock()
+	defer vaultLock.Unlock()
+
 	if err := vault.AddTestLogicalBackend("pki", pki.Factory); err != nil {
```

Comment:
This call mutates a global, and when more than one vault-related test is run in parallel they race.